tools/update-db.sh

```diff
@@ -14,18 +15,23 @@ source /etc/makepkg.conf
 cd "$ROOT"/out
 curl -fOs "https://arch.osamc.de/proaudio/${CARCH}/proaudio.{db,files}.tar.gz"
 
-# Add newly built packages to db
-repo-add proaudio.db.tar.gz *$PKGEXT
-
 # Cleanup old packages from db
-> "$TMP"/removal
+REMOVAL=()
 for PKGPATH in $(bsdtar -tf proaudio.db.tar.gz '*/desc'); do
     FILENAME="$(bsdtar -xOqf proaudio.db.tar.gz "$PKGPATH" | sed -n '/^%FILENAME%$/ {n;p}')"
     if ! grep -qxF "$FILENAME" "$TMP"/packagelist; then
         PKGNAME="$(bsdtar -xOqf proaudio.db.tar.gz "$PKGPATH" | sed -n '/^%NAME%$/ {n;p}')"
-        echo "$FILENAME" >> "$TMP"/removal
-        repo-remove proaudio.db.tar.gz "$PKGNAME"
+        REMOVAL+=(${PKGNAME})
     fi
 done
```

Comment:
Dieser Code ist, wie schon im Chat erwähnt, m.E. a) ineffizient (wiederholtes `tar -x`) und b) schwer zu verstehen.

Ich würde vorschlagen, die Stelle noch mal zu refaktorieren und ggf. mit Kommentaren zu versehen, was die Intention der einzelnen Kommandos / Blöcke ist.

Reply:
Hoffe es ist jetzt verständlicher :) 348fa8105b2f6286b7742f6bc3dd7d09003bdc4d

Vielleicht gibt es ja irgendwann ein release von [dbscripts](https://gitlab.archlinux.org/archlinux/dbscripts) und ein paar mehr Tools, mit denen man solche dinge einfacher machen kann.

Reply:
Vlt kurz zum Hintergrund dieser update-db: das Ziel ist, die db aus dem Live-Repo als Cache/Basis zu benutzen und nur geänderte Pakete auch hinzuzufügen. Um die möglichen edge cases wie Paket entfernen/umbenennen, Multi-Pakete zusammenführen etc. auch zu unterstützen, müssen wir in dem Skript halt berücksichtigen, dass es alte Pakete in der db geben kann und diese löschen.

Reply:
> Tbh, I think this is were I personally would ditch the shell and write a Python script instead. ;)

Well, this is Arch, not Ubuntu (wasn't it KISS = "keep it shell-scripted"? 😜)